apps/web/components/I18nLanguageHandler.tsx

```diff
@@ -22,16 +22,19 @@ export function useViewerI18n() {
  */
 const I18nLanguageHandler = (): null => {
   const { i18n } = useTranslation("common");
-  const locale = useViewerI18n().data?.locale || "en";
+  const viewerI18n = useViewerI18n();
+  const locale = viewerI18n.data?.locale || i18n.language;
 
   useEffect(() => {
-    if (locale && i18n.language && i18n.language !== locale) {
-      if (typeof i18n.changeLanguage === "function") i18n.changeLanguage(locale);
+    // bail early when i18n = {}
+    if (isEmpty(i18n)) return;
```

Comment:
Do we really want to roll our own utility library when we've got lodash treeshaking working? 🤔